### ISSUE TITLE
Animation debug_assert failure

### ISSUE THREAD

**Issue by MartinodF** [servo/servo#12171]
called `Result::unwrap()` on an `Err` value: RecvError
### URL:
https://www.lays.it/piacere-lays/full
### Backtrace:
```
frame #0  - 0x0000557bb5f3ad8d - <unknown>
frame #1  - 0x0000557bb5f3ad15 - <unknown>
frame #2  - 0x0000557bb5108359 - <unknown>
frame #3  - 0x0000557bb5f31b98 - <unknown>
frame #4  - 0x0000557bb603914c - <unknown>
frame #5  - 0x0000557bb6053761 - <unknown>
frame #6  - 0x0000557bb603a9ba - <unknown>
frame #7  - 0x0000557bb60536fe - <unknown>
frame #8  - 0x0000557bb6089cff - <unknown>
frame #9  - 0x0000557bb50d320d - <unknown>
frame #10 - 0x0000557bb51665f8 - <unknown>
frame #11 - 0x0000557bb514bad7 - <unknown>
frame #12 - 0x0000557bb5106a47 - <unknown>
frame #13 - 0x0000557bb605d95b - <unknown>
frame #14 - 0x0000557bb605d8fe - <unknown>
frame #15 - 0x0000557bb5107cbd - <unknown>
frame #16 - 0x0000557bb6051924 - <unknown>
frame #17 - 0x00007fc253398483 - start_thread
frame #18 - 0x00007fc252ec16dc - clone
frame #19 - 0x0000000000000000 - <unknown>

```

**Comment by MartinodF** [comment]
The same page, with a debug build compiled from master a few minutes ago (commit `1ed06777ee1e69c8b56715c897a5bd66ad9e235e`), shows a different error:

```
/mach run -d -- https://www.lays.it/piacere-lays/full
thread 'LayoutWorker worker 2/3' panicked at 'assertion failed: !state.expired', /home/mart/Repos/servo/components/style/animation.rs:533
stack backtrace:
   1:     0x55915b288a1f - std::sys::backtrace::tracing::imp::write::h6528da8103c51ab9
   2:     0x55915b29064b - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hbe741a5cc3c49508
   3:     0x55915b2902af - std::panicking::default_hook::he0146e6a74621cb4
   4:     0x55915aefa2aa - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h35c053fbde45511b
                        at /home/mart/Repos/servo/components/util/panicking.rs:45
   5:     0x55915aefa169 - _<std..thread..local..LocalKey<T>>::with::he5a18532e611f086
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/thread/local.rs:211
   6:     0x55915aef9fe6 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h7cb4a9718444f10d
                        at /home/mart/Repos/servo/components/util/panicking.rs:40
   7:     0x55915b2760ec - std::panicking::rust_panic_with_hook::h983af77c1a2e581b
   8:     0x559157b685b7 - std::panicking::begin_panic::h0bf39f6d43ab9349
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:328
   9:     0x559157dc66e2 - style::animation::update_style_for_animation::h1db51cedeae7ea1b
                        at /home/mart/Repos/servo/components/servo/<std macros>:3
  10:     0x559157da3577 - style::matching::PrivateMatchMethods::update_animations_for_cascade::hccfc4d1322dfb749
                        at /home/mart/Repos/servo/components/style/matching.rs:501
  11:     0x559157da24cc - style::matching::PrivateMatchMethods::cascade_node_pseudo_element::ha297041b65e9e83b
                        at /home/mart/Repos/servo/components/style/matching.rs:388
  12:     0x559157e08048 - style::matching::MatchMethods::cascade_node::_$u7b$$u7b$closure$u7d$$u7d$::h3b519f5670b818c1
  13:     0x559157e07de1 - style::selector_impl::SelectorImplExt::each_eagerly_cascaded_pseudo_element::_$u7b$$u7b$closure$u7d$$u7d$::hc49b7ee3c52b047d
  14:     0x559157e07d21 - _<style..selector_impl..ServoSelectorImpl as style..selector_impl..SelectorImplExt>::each_pseudo_element::hc040641c45566960
                        at /home/mart/Repos/servo/components/style/selector_impl.rs:253
  15:     0x559157e07cf3 - style::selector_impl::SelectorImplExt::each_eagerly_cascaded_pseudo_element::h74955c739b6bb45a
                        at /home/mart/Repos/servo/components/style/selector_impl.rs:79
  16:     0x559157da206d - style::matching::MatchMethods::cascade_node::he54594f24c909660
                        at /home/mart/Repos/servo/components/style/matching.rs:713
  17:     0x559157d8d1ea - style::traversal::recalc_style_at::h15c7c808807ca35a
                        at /home/mart/Repos/servo/components/style/traversal.rs:221
  18:     0x559157d8bd67 - _<layout..traversal..RecalcStyleAndConstructFlows<'lc> as style..traversal..DomTraversalContext<N>>::process_preorder::ha9c74f3a0dd5c895
                        at /home/mart/Repos/servo/components/layout/traversal.rs:78
  19:     0x559157e6f1a0 - style::parallel::top_down_dom::heb6002f145deaadd
                        at /home/mart/Repos/servo/components/style/parallel.rs:67
  20:     0x559157be0acb - _<util..workqueue..WorkerThread<QueueData, WorkData>>::start::heaa899404d32e758
                        at /home/mart/Repos/servo/components/util/workqueue.rs:182
  21:     0x559157bdf95f - _<util..workqueue..WorkQueue<QueueData, WorkData>>::new::_$u7b$$u7b$closure$u7d$$u7d$::h700ecdbcd68efe5d
  22:     0x559157bdf815 - _<std..panic..AssertUnwindSafe<F> as core..ops..FnOnce<()>>::call_once::h7c5c4ec8301a7f08
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panic.rs:251
  23:     0x559157bdf728 - std::panicking::try::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h0dbee496992bb7c0
  24:     0x559157bdffb4 - std::panicking::try::call::h4aecc47b24882913
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:272
  25:     0x55915b29aa8b - __rust_try
  26:     0x55915b29aa2e - __rust_maybe_catch_panic
  27:     0x559157bdf67d - std::panicking::try::_$u7b$$u7b$closure$u7d$$u7d$::hde1ee724dc352c95
  28:     0x559157bdf5cf - _<std..thread..local..LocalKey<T>>::with::h875d55103c36eb37
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/thread/local.rs:211
  29:     0x559157bdf419 - std::panicking::try::hd0682edce1d68070
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:235
  30:     0x559157bdf2af - std::panic::catch_unwind::hfb341bb5feae5200
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panic.rs:307
  31:     0x559157bdf0d0 - std::thread::Builder::spawn::_$u7b$$u7b$closure$u7d$$u7d$::hb03a0345a9478a0e
  32:     0x559157be01e1 - _<F as alloc..boxed..FnBox<A>>::call_box::h21cccc509f64f626
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/liballoc/boxed.rs:544
  33:     0x55915b28ea54 - std::sys::thread::Thread::new::thread_start::h9c883b6d445ece46
  34:     0x7fccb8dba483 - start_thread
  35:     0x7fccb88e36dc - clone
  36:                0x0 - <unknown>
```

**Comment by jdm** [comment]
cc @emilio it appears some debug asserts in update_style_for_animation are being hit.

**Comment by jdm** [comment]
The release build panic is tracked in #12151.

**Comment by dzbarsky** [comment]
I got a different error on this page in debug servo on OSX
```
➜  servo git:(master) ✗ ./mach run -d https://www.lays.it/piacere-lays/full
thread 'LayoutWorker worker 1/6' panicked at 'assertion failed: !state.expired', /Users/dzbarsky/mozilla/servo/components/style/animation.rs:533
stack backtrace:
   1:        0x10de8ba4b - std::sys::backtrace::tracing::imp::write::h6528da8103c51ab9
   2:        0x10de91fea - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hbe741a5cc3c49508
   3:        0x10de91bfa - std::panicking::default_hook::he0146e6a74621cb4
   4:        0x10d9c309a - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h8b3395801da9233f
   5:        0x10d9c2f63 - _<std..thread..local..LocalKey<T>>::with::hf5850e9f85f835d5
   6:        0x10d9c2db6 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hdf03bb4999f2308b
   7:        0x10de794d4 - std::panicking::rust_panic_with_hook::h983af77c1a2e581b
   8:        0x10a5cac07 - std::panicking::begin_panic::h0bf39f6d43ab9349
   9:        0x10a83d5c2 - style::animation::update_style_for_animation::hb56aaf9ff70f7a22
  10:        0x10a81a027 - style::matching::PrivateMatchMethods::update_animations_for_cascade::h5de968474ac29a80
  11:        0x10a818f7c - style::matching::PrivateMatchMethods::cascade_node_pseudo_element::h6ed50833db4fd171
  12:        0x10a87fe88 - style::matching::MatchMethods::cascade_node::_$u7b$$u7b$closure$u7d$$u7d$::h492896281a882a90
  13:        0x10a87fc21 - style::selector_impl::SelectorImplExt::each_eagerly_cascaded_pseudo_element::_$u7b$$u7b$closure$u7d$$u7d$::h9fc2713bb69fcfa7
  14:        0x10a87fb61 - _<style..selector_impl..ServoSelectorImpl as style..selector_impl..SelectorImplExt>::each_pseudo_element::hec75134dbc46091b
  15:        0x10a87fb2b - style::selector_impl::SelectorImplExt::each_eagerly_cascaded_pseudo_element::hc96b7fb1a61f8876
  16:        0x10a818b0d - style::matching::MatchMethods::cascade_node::hc0964f95c6402c4c
  17:        0x10a80499a - style::traversal::recalc_style_at::h3836cbbf1cced576
  18:        0x10a803517 - _<layout..traversal..RecalcStyleAndConstructFlows<'lc> as style..traversal..DomTraversalContext<N>>::process_preorder::h588d2184aa33d145
  19:        0x10a8e85de - style::parallel::top_down_dom::hfbb4a75348133cd6
  20:        0x10a646248 - _<util..workqueue..WorkerThread<QueueData, WorkData>>::start::h34aa2175414961ea
  21:        0x10a645181 - _<util..workqueue..WorkQueue<QueueData, WorkData>>::new::_$u7b$$u7b$closure$u7d$$u7d$::h52af97a69e452536
  22:        0x10a64504f - _<std..panic..AssertUnwindSafe<F> as core..ops..FnOnce<()>>::call_once::hd271a0880ce42a9c
  23:        0x10a644f98 - std::panicking::try::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hf0d548f2e370888b
  24:        0x10a645714 - std::panicking::try::call::hcffa1cd9ff3e95d8
  25:        0x10de952ab - __rust_try
  26:        0x10de95245 - __rust_maybe_catch_panic
  27:        0x10a644eed - std::panicking::try::_$u7b$$u7b$closure$u7d$$u7d$::h0b191d640300b8cd
  28:        0x10a644e3f - _<std..thread..local..LocalKey<T>>::with::h9c14260141e7529c
  29:        0x10a644c79 - std::panicking::try::ha48f74ec899df47d
  30:        0x10a644b5e - std::panic::catch_unwind::hed745db425ac80d2
  31:        0x10a6449ae - std::thread::Builder::spawn::_$u7b$$u7b$closure$u7d$$u7d$::hac3f6daeaba38307
  32:        0x10a645934 - _<F as alloc..boxed..FnBox<A>>::call_box::h3534e14420f9533c
  33:        0x10de90d28 - std::sys::thread::Thread::new::thread_start::h9c883b6d445ece46
  34:     0x7fff9eabbc12 - _pthread_body
  35:     0x7fff9eabbb8f - _pthread_start
thread 'LayoutWorker worker 5/6' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', ../src/libcore/result.rs:785
stack backtrace:
   1:        0x10de8ba4b - std::sys::backtrace::tracing::imp::write::h6528da8103c51ab9
   2:        0x10de91fea - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hbe741a5cc3c49508
   3:        0x10de91bfa - std::panicking::default_hook::he0146e6a74621cb4
   4:        0x10d9c309a - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h8b3395801da9233f
   5:        0x10d9c2f63 - _<std..thread..local..LocalKey<T>>::with::hf5850e9f85f835d5
   6:        0x10d9c2db6 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hdf03bb4999f2308b
   7:        0x10de794d4 - std::panicking::rust_panic_with_hook::h983af77c1a2e581b
   8:        0x10de925c6 - std::panicking::begin_panic::he426e15a3766089a
   9:        0x10de7ab18 - std::panicking::begin_panic_fmt::hdddb415186c241e7
  10:        0x10de9221f - rust_begin_unwind
  11:        0x10deb90a0 - core::panicking::panic_fmt::hf4e16cb7f0d41a25
  12:        0x10a81a4d5 - core::result::unwrap_failed::h7ae144e7dd8bd475
  13:        0x10a81a1a8 - _<core..result..Result<T, E>>::unwrap::h2abd9d70b27c9fbc
  14:        0x10a819df5 - style::matching::PrivateMatchMethods::update_animations_for_cascade::h5de968474ac29a80
  15:        0x10a818f7c - style::matching::PrivateMatchMethods::cascade_node_pseudo_element::h6ed50833db4fd171
  16:        0x10a818a28 - style::matching::MatchMethods::cascade_node::hc0964f95c6402c4c
  17:        0x10a80499a - style::traversal::recalc_style_at::h3836cbbf1cced576
  18:        0x10a803517 - _<layout..traversal..RecalcStyleAndConstructFlows<'lc> as style..traversal..DomTraversalContext<N>>::process_preorder::h588d2184aa33d145
  19:        0x10a8e85de - style::parallel::top_down_dom::hfbb4a75348133cd6
  20:        0x10a646248 - _<util..workqueue..WorkerThread<QueueData, WorkData>>::start::h34aa2175414961ea
  21:        0x10a645181 - _<util..workqueue..WorkQueue<QueueData, WorkData>>::new::_$u7b$$u7b$closure$u7d$$u7d$::h52af97a69e452536
  22:        0x10a64504f - _<std..panic..AssertUnwindSafe<F> as core..ops..FnOnce<()>>::call_once::hd271a0880ce42a9c
  23:        0x10a644f98 - std::panicking::try::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hf0d548f2e370888b
  24:        0x10a645714 - std::panicking::try::call::hcffa1cd9ff3e95d8
  25:        0x10de952ab - __rust_try
  26:        0x10de95245 - __rust_maybe_catch_panic
  27:        0x10a644eed - std::panicking::try::_$u7b$$u7b$closure$u7d$$u7d$::h0b191d640300b8cd
  28:        0x10a644e3f - _<std..thread..local..LocalKey<T>>::with::h9c14260141e7529c
  29:        0x10a644c79 - std::panicking::try::ha48f74ec899df47d
  30:        0x10a644b5e - std::panic::catch_unwind::hed745db425ac80d2
  31:        0x10a6449ae - std::thread::Builder::spawn::_$u7b$$u7b$closure$u7d$$u7d$::hac3f6daeaba38307
  32:        0x10a645934 - _<F as alloc..boxed..FnBox<A>>::call_box::h3534e14420f9533c
  33:        0x10de90d28 - std::sys::thread::Thread::new::thread_start::h9c883b6d445ece46
  34:     0x7fff9eabbc12 - _pthread_body
  35:     0x7fff9eabbb8f - _pthread_start
```

**Comment by emilio** [comment]
I'm looking at it, the problem in #12151 is a different problem and should be fixed by #12159
